arangod/Aql/Functions.cpp

```diff
@@ -4569,6 +4606,158 @@ AqlValue Functions::RandomToken(ExpressionContext*, transaction::Methods*,
   return AqlValue(generator.random(static_cast<size_t>(length)));
 }
 
+/// @brief function IP4_FROM_NUMBER
+AqlValue Functions::Ip4FromNumber(ExpressionContext* expressionContext, transaction::Methods*,
+                                  VPackFunctionParameters const& parameters) {
+  static char const* AFN = "IP4_FROM_NUMBER";
+
+  AqlValue const& value = extractFunctionParameterValue(parameters, 0);
+
+  if (!value.isNumber()) {
+    registerWarning(expressionContext, AFN, TRI_ERROR_QUERY_FUNCTION_ARGUMENT_TYPE_MISMATCH);
+    return AqlValue(AqlValueHintNull());
+  }
+  
+  int64_t input = value.toInt64();
+  if (input < 0 || static_cast<uint64_t>(input) > UINT32_MAX) {
+    registerInvalidArgumentWarning(expressionContext, AFN);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  uint64_t number = static_cast<uint64_t>(input);
+  
+  // in theory, we only need a 15 bytes buffer here, as the maximum result
+  // string is "255.255.255.255"
+  char result[32]; 
+
+  char* p = &result[0];
+  // first part
+  uint64_t digit = (number & 0xff000000ULL) >> 24ULL; 
+  p += basics::StringUtils::itoa(digit, p);
+  *p++ = '.';
+  // second part
+  digit = (number & 0x00ff0000ULL) >> 16ULL; 
+  p += basics::StringUtils::itoa(digit, p);
+  *p++ = '.';
+  // third part
+  digit = (number & 0x0000ff00ULL) >> 8ULL; 
+  p += basics::StringUtils::itoa(digit, p);
+  *p++ = '.';
+  // fourth part
+  digit = (number & 0x0000ffULL); 
+  p += basics::StringUtils::itoa(digit, p);
```

Comment:
Can't we use an existing function like `inet_ntoa(3)` or `inet_ntop(3)`?

Reply:
done